src/components/CurrentScenarioSelector/CurrentScenarioSelector.js

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { HierarchicalComboBox } from '@cosmotech/ui';
+import { useSortedScenarioList } from '../../hooks/ScenarioListHooks';
+import { STATUSES } from '../../state/commons/Constants';
+import {
+  useCurrentScenarioData,
+  useCurrentScenarioReducerStatus,
+  useFindScenarioById,
+} from '../../state/hooks/ScenarioHooks';
+import { t } from 'i18next';
+
+const CurrentScenarioSelector = ({ disabled, renderInputToolTip }) => {
+  const sortedScenarioList = useSortedScenarioList();
+  const currentScenarioData = useCurrentScenarioData();
+  const currentScenarioStatus = useCurrentScenarioReducerStatus();
+
+  const changeScenario = useFindScenarioById();
+
+  const handleScenarioChange = useCallback(
+    (event, scenario) => {
+      changeScenario(scenario.id);
+    },
+    [changeScenario]
+  );
+
+  const noScenario = currentScenarioData === null || sortedScenarioList.length === 0;
+  const isSwitchingScenario = currentScenarioStatus === STATUSES.LOADING;
+
+  const scenarioListLabel = noScenario ? null : t('views.scenario.dropdown.scenario.label', 'Scenario');
+  const scenarioValidationStatusLabels = {
+    rejected: t('views.scenario.validation.rejected', 'Rejected'),
+    validated: t('views.scenario.validation.validated', 'Validated'),
+  };
+
+  const hierarchicalComboBoxLabels = {
+    label: scenarioListLabel,
+    validationStatus: scenarioValidationStatusLabels,
+  };
+
+  return (
+    <HierarchicalComboBox
+      value={currentScenarioData}
+      values={sortedScenarioList}
+      labels={hierarchicalComboBoxLabels}
+      renderInputToolType={renderInputToolTip}
+      disabled={disabled || noScenario || isSwitchingScenario}
+      handleChange={handleScenarioChange}
+    />
+  );
+};
+
+CurrentScenarioSelector.propTypes = {
+  /**
+   * Component's labels:
+   * Structure:
+   * <pre>
+     {
+       label: 'string',
+       validationStatus:
+       {
+         rejected: 'string',
+         validated: 'string',
+       }
+     }
+   * </pre>
+   */
```

Comment:
Comment seems a bit overkill (object just after tell type by himself)

Reply:
in fact, this prop is not used ...
Remove prop, done